### DISCUSSION TITLE
Difference to ext-php-rs

### DISCUSSION THREAD

**Discussion by nikeee** [phper-framework/phper#60] [Q&A]
If I'm getting this right, this library look like it solves a slimilar problem as [ext-php-rs](https://github.com/davidcole1340/ext-php-rs).

As I'm unfarmiliar in PHP extension development and would like to play around a bit, what are the differences between phper and ext-php-rs? Are there use cases where one is more suitable than the other (PHP version support, general feature set, performance, portability, general project goal, etc.)?

phper
- Docs: https://docs.rs/phper
- Examples: https://github.com/phper-framework/phper/tree/master/examples

ext-php-rs:
- Docs: https://docs.rs/ext-php-rs
- Guide: https://davidcole1340.github.io/ext-php-rs/introduction.html

**Comment by jmjoy** [comment]
It seems that phper is earlier than ext-php-rs, so you can choose by yourself.

**Reply by sashaaro** [reply]
@jmjoy is phper support zts? by code I saw mention and it seems yes,
 but didn't see any description about it and tested support zts section unchecked in the readme

**Reply by jmjoy** [reply]
Not support yet.

**Comment by joehoyle** [comment]
Hi @jmjoy I'm a contributor and somewhat-heavy use of ext-php-rs, and jst discovered phper too, thanks for creating the project! I wondered if you could share your thoughts on the two approaches of the projects. Is it possible to bring the projects together? For one, I'd prefer to be putting contributions efforts as much combined as possible, and would be open to pulling the projects together if that would be possible.

**Reply by jmjoy** [reply]
This is a framework that supports the development of PHP extensions with pure Rust (possibly without C/C++), can effectively use the Rust ecosystem, mainly the encapsulation of Zend api. 
 
I see this enthusiastic person made a comparison: <https://github.com/rtio/rust_php_ext_examples>. 
 
I don't think it's necessary to merge the two projects, because many open source projects are reinventing the wheel. :smile:
